src/WebJobs.Script/Workers/Rpc/JobHostRpcWorkerChannelManager.cs

```diff
@@ -12,45 +13,86 @@ namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
     internal class JobHostRpcWorkerChannelManager : IJobHostRpcWorkerChannelManager
     {
         private readonly ILogger _logger;
-        private ConcurrentDictionary<string, IRpcWorkerChannel> _channels = new ConcurrentDictionary<string, IRpcWorkerChannel>();
+        private ConcurrentDictionary<string, IRpcWorkerChannelHolder> _channels = new ConcurrentDictionary<string, IRpcWorkerChannelHolder>();
 
         public JobHostRpcWorkerChannelManager(ILoggerFactory loggerFactory)
         {
             _logger = loggerFactory.CreateLogger<JobHostRpcWorkerChannelManager>();
         }
 
-        public void AddChannel(IRpcWorkerChannel channel)
+        public void AddChannel(IRpcWorkerChannel channel, string language)
         {
-            _channels.TryAdd(channel.Id, channel);
+            if (_channels.TryGetValue(language, out IRpcWorkerChannelHolder channels))
+            {
+                channels.TryAdd(channel.Id, channel);
+            }
+            else
+            {
+                _channels.TryAdd(language, new IRpcWorkerChannelHolder
+                {
+                    [channel.Id] = channel,
+                });
+            }
         }
 
         public Task<bool> ShutdownChannelIfExistsAsync(string channelId, Exception workerException)
         {
-            if (_channels.TryRemove(channelId, out IRpcWorkerChannel removedChannel))
+            foreach (string language in _channels.Keys)
             {
-                _logger.LogDebug("Disposing JobHost language worker channel with id:{workerId}", removedChannel.Id);
-                removedChannel.TryFailExecutions(workerException);
-                (removedChannel as IDisposable)?.Dispose();
-                return Task.FromResult(true);
+                if (_channels.TryGetValue(language, out IRpcWorkerChannelHolder channels))
+                {
+                    if (channels.TryRemove(channelId, out IRpcWorkerChannel rpcChannel))
+                    {
+                        _logger.LogDebug("Disposing language worker channel with id:{workerId}", rpcChannel.Id);
+                        rpcChannel.TryFailExecutions(workerException);
+                        (rpcChannel as IDisposable)?.Dispose();
+                        return Task.FromResult(true);
+                    }
+                }
             }
             return Task.FromResult(false);
         }
 
         public void ShutdownChannels()
         {
-            foreach (string channelId in _channels.Keys)
+            foreach (string language in _channels.Keys)
             {
-                if (_channels.TryRemove(channelId, out IRpcWorkerChannel removedChannel))
+                if (_channels.TryRemove(language, out IRpcWorkerChannelHolder channels))
```

Comment:
nit: renaming language to languageChannelId and Channels to RemovedChannels

Reply:
`languageChannelId` doesn't makes sense here, it's the language only and can have values like "java", "node", etc.  Renaming `removedChannels`.